dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -54,34 +56,50 @@
     public ListenableFuture<Response> execute(Endpoint endpoint, Request request) {
         SettableFuture<Response> future = SettableFuture.create();
 
-        Supplier<ListenableFuture<Response>> callSupplier = () -> delegate.execute(endpoint, request);
-        FutureCallback<Response> retryer = new RetryingCallback<>(callSupplier, future);
-        Futures.addCallback(callSupplier.get(), retryer, DIRECT_EXECUTOR);
+        Function<Integer, ListenableFuture<Response>> callSupplier = attempt -> {
+            // TODO(dfox): include retry number in the request somehow
+            return delegate.execute(endpoint, request);
+        };
+        FutureCallback<Response> retryer = new RetryingCallback(callSupplier, future);
+        Futures.addCallback(callSupplier.apply(0), retryer, DIRECT_EXECUTOR);
```

Comment:
```suggestion
        return DialogueFutures.addDirectCallback(callSupplier.apply(0), retryer);
```

Reply:
If it's OK with you I think I'd actually prefer to just stick with the vanilla guava - I find it kinda reassuring that there's no magic going on under the hood

Reply:
It just seems like extra boilerplate 🤷‍♂ feels off to have a utility exactly for this and then not use it